src/PromoService.kt

```
@@ -0,0 +1,1342 @@
+            try {
+                val updatePromoRequest = objectMapper.convert<UpdatePromoRequest>(promo.promoCodeMetaDto)!!
+                updatePromoRequest.userSettingsDetailsDto?.validate()
+                updatePromo(promo.code, updatePromoRequest)
+                updatedPromoCount += 1
+                logger.info {
+                    "Updated the promo ${promo.code} with campaignType as ${updatePromoRequest.campaignType} " +
+                            "and user settings details -: eligible client type as ${updatePromoRequest.userSettingsDetailsDto?.eligibleClientType}, " +
+                            "grace period equal to ${updatePromoRequest.userSettingsDetailsDto?.gracePeriod ?: 0} days, with fulfillment rule as ${updatePromoRequest.userSettingsDetailsDto?.fulfillmentRule} " +
+                            "and fulfillment period equal to ${updatePromoRequest.userSettingsDetailsDto?.fulfillmentPeriod} days."
+                }
+            } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)